--- a/trunk/tools/mktplinkfw/mktplinkfw2.c
+++ b/trunk/tools/mktplinkfw/mktplinkfw2.c
@@ -147,6 +147,12 @@ static struct flash_layout layouts[] = {
 		.kernel_la	= 0x80000000,
 		.kernel_ep	= 0x80000000,
 		.rootfs_ofs	= 0x140000,
+	}, {
+		.id		= "16Mmtk",
+		.fw_max_len	= 0xfa0000,
+		.kernel_la	= 0x80000000,
+		.kernel_ep	= 0x80000000,
+		.rootfs_ofs	= 0x140000,
 	}, {
 		.id		= "8MSUmtk", /* Split U-Boot OS */
 		.fw_max_len	= 0x770000,
