--- a/trunk/linux-3.4.x/drivers/char/ralink/ralink_gpio.c
+++ b/trunk/linux-3.4.x/drivers/char/ralink/ralink_gpio.c
@@ -1518,7 +1518,7 @@ int __init ralink_gpio_init(void)
 	gpiomode = le32_to_cpu(*(volatile u32 *)(RALINK_REG_GPIOMODE));
 	gpiomode &= ~(RALINK_GPIOMODE_DFT);		// clear bit[2:4] UARTF_SHARE_MODE
 #if defined (CONFIG_RALINK_MT7620)
-	gpiomode &= ~(RALINK_GPIOMODE_WLED);		// clear bit[13] WLAN_LED
+	gpiomode |= RALINK_GPIOMODE_WLED;		// clear bit[13] WLAN_LED
 #endif
 #if defined (CONFIG_RALINK_MT7620)
 	gpiomode &= ~(RALINK_GPIOMODE_SPI_REFCLK);	// clear bit[12] SPI_REFCLK0_MODE
