--- a/trunk/user/httpd/ralink.c
+++ b/trunk/user/httpd/ralink.c
@@ -1485,7 +1485,7 @@ ej_wl_scan_xg(const char * net_device, int eid, webs_t wp, int argc, char **argv
 	char ssid_str[128];
 	char site_line[SSURV_LINE_LEN_MAX+1];
 	char site_chnl[4];
-	char site_ssid[34];
+	char site_ssid[65];
 	char site_bssid[24];
 	char site_signal[10];
 	struct iwreq wrq;
@@ -1553,13 +1553,13 @@ ej_wl_scan_xg(const char * net_device, int eid, webs_t wp, int argc, char **argv
 			memcpy(site_line, sp, line_len);
 
 			memcpy(site_chnl, sp+x_ch, 3);
-			memcpy(site_ssid, sp+x_ssid, 33);
+			memcpy(site_ssid, sp+x_ssid, 64);
 			memcpy(site_bssid, sp+x_bssid, 20);
 			memcpy(site_signal, sp+x_signal, 9);
 
 			site_line[line_len] = '\0';
 			site_chnl[3] = '\0';
-			site_ssid[33] = '\0';
+			site_ssid[65] = '\0';
 			site_bssid[20] = '\0';
 			site_signal[9] = '\0';
 
